eZ/Publish/Core/MVC/Symfony/Locale/UserLanguagePreferenceProvider.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\Core\MVC\Symfony\Locale;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Provides list of user-preferred languages.
+ */
+class UserLanguagePreferenceProvider
```

Comment:
We should extract `UserLanguagePreferenceProviderinterface`.  IMHO `\eZ\Publish\Core\MVC\Symfony\View\Builder\ViewBuilder` implementations in `repository-forms` / `ezplatform-admin-ui` shouldn't depends on this concrete implementation, but rather on contract exposed from `ezpublish-kernel`.